website/server/controllers/api-v4/members.js

```diff
@@ -0,0 +1,49 @@
+import {model as User} from '../../models/user';
+import {authWithHeaders} from '../../middlewares/auth';
+
+let api = {};
+
+
+api.getUsernameAutocompletes = {
+  method: 'GET',
+  url: '/members/find/:username',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    res.set('Cache-Control', 'public, max-age=300000'); // 5 minutes
```

Comment:
I fee like now that we have different results based on which user makes the request this is going to create some problems

Reply:
Oh, that's a good point. I could try if it works to only cache some of the calls? So that only the public searches are cached and all searches in private spaces are not cached. Do you know if that will work with googles caching?

Reply:
So we can cache everything that's the same for all users, everything else should not be cached. So I guess if the results for public searches is the same for everyone it'd be okay to cache them